### ISSUE TITLE
[Server Component] TypeError: Cannot read properties of undefined (reading 'link')

### ISSUE THREAD

**Issue by kingyue737** [nuxt/nuxt#27491]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.20.3
- Nuxt Version:     3.11.2
- CLI Version:      3.11.1
- Nitro Version:    2.9.6
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      devtools
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/github-ndhpvf

### Describe the bug

Click the button to toggle prop `text` of server component `MyComponent`.

The first time, it renders correctly. But when I click once more. It shows the following error:
```
TypeError: Cannot read properties of undefined (reading 'link')
```

It seems to be a cache error

### Additional context

If I remove `Math.random()`, no error occurs.

Can also reproduce locally:

------------------------------
- Operating System: Windows 11
- Node Version:     v20.13.0
- Nuxt Version:     3.11.2
- CLI Version:      3.11.1
- Nitro Version:    2.9.6
- Package Manager:  pnpm@9.2.0
------------------------------

**Comment by hoanghiep1x0** [comment]
This seems to be an error because you are using a variable without data in the component file.

**Comment by danielroe** [comment]
This is resolved in the latest Nuxt version (v3.13.1).
